keep/providers/vectordev_provider/vectordev_provider.py

```diff
@@ -32,33 +40,22 @@ def validate_config(self):
         )
 
     def _format_alert(
-        event: list[dict], provider_instance: "BaseProvider" = None
+        event: dict, provider_instance: "BaseProvider" = None
     ) -> AlertDto | list[AlertDto]:
-        events = []
-        # event is a list of events
-        for e in event:
-            event_json = None
-            try:
-                event_json = json.loads(e.get("message"))
-            except json.JSONDecodeError:
-                pass
+        event_type = event["source_type"]
+        provider_class = ProvidersFactory.get_provider_class(VectordevProvider.SOURCE_TO_PROVIDER_MAP[event_type])
+        return provider_class._format_alert(event["event"],provider_instance)
 
-            events.append(
-                AlertDto(
-                    name="",
-                    host=e.get("host"),
-                    message=e.get("message"),
-                    description=e.get("message"),
-                    lastReceived=e.get("timestamp"),
-                    source_type=e.get("source_type"),
-                    source=["vectordev"],
-                    original_event=event_json,
-                )
-            )
```

Comment:
@sainad2222 would leave this in if 
`provider_class = ProvidersFactory.get_provider_class(provider)` didn't find the provider (by `event_type`).